libs/auth/routing/src/redirect.module.ts

```diff
@@ -1,13 +1,59 @@
-import { NgModule } from '@angular/core';
+import {
+  NgModule,
+  inject,
+} from '@angular/core';
+import {
+  ActivatedRoute,
+  Router,
+} from '@angular/router';
 import { EffectsModule } from '@ngrx/effects';
+import {
+  from,
+  of,
+} from 'rxjs';
 
+import {
+  DAFF_AUTH_UNAUTHENTICATED_HOOKS,
+  DaffAuthActionTypes,
+  DaffAuthLoginActionTypes,
+  DaffAuthUnauthenticatedHook,
+} from '@daffodil/auth/state';
+
+import { DAFF_AUTH_ROUTING_CONFIG } from './config/public_api';
 import { DaffAuthRedirectEffects } from './effects/redirect.effects';
 
+
 @NgModule({
   imports: [
     EffectsModule.forFeature([
       DaffAuthRedirectEffects,
     ]),
   ],
+  providers: [
+    {
+      provide: DAFF_AUTH_UNAUTHENTICATED_HOOKS,
+      multi: true,
+      useFactory: () => {
```

Comment:
Can you extract this to a separate function?

Reply:
done